todoman/ui.py

```diff
@@ -251,9 +251,23 @@ def detailed(self, todo):
         return rv
 
     def format_date(self, date):
+        """
+        Returns date in the following format:
+        * if date == today or tomorrow: "Today" or "Tomorrow"
+        * else: return a string representing that date
+        * if no date is supplied, it returns empty_date
+        """
         if date:
-            rv = date.strftime(self.date_format)
-            return rv
+            date_tomorrow = self.now.date() + timedelta(days=1)
+            # Get a dictionary which stores the dates for which
+            # we don't have to merely return a date string and
+            # map it to the special string we need to return
+            WIDTH = len(date.strftime(self.date_format))
```

Comment:
I guess you would want something like this:
```
def __init__(self, date_format):
    self.date_format = date_format
    self._localtimezone = tzlocal()
    self.now = datetime.now().replace(tzinfo=self._localtimezone)
    # Get a dictionary which stores the dates for which
    # we don't have to merely return a date string and
    # map it to the special string we need to return
    self.special_dates = {
        self.now.date(): "Today".rjust(self.date_width, " "),
        date_tomorrow: "Tomorrow".rjust(self.date_width, " "),
    }
    # An empty date which should be used in case no date is present
    self.empty_date = " " * len(self.format_date(self.now))
    self.date_width = len(self.empty_date)
    self._parsedatetime_calendar = parsedatetime.Calendar()
```
But then wouldn't  ```self.empty_date``` call ```self.format_date```, where ```self.date_width``` wouldn't be present and there would be an exception.
Should we find a different way for calculating ```self.empty_date```  or should we leave date_width in function itself?